python/ray/external_storage.py

```diff
@@ -243,6 +256,43 @@ def delete_spilled_objects(self, urls: List[str]):
             filename = parse_url_with_offset(url.decode()).base_url
             os.remove(os.path.join(self.directory_path, filename))
 
+    def destroy_external_storage(self):
```

Comment:
Can we just delete the entire directory with one call?

Reply:
Agree this is a simpler logic; The only concern I have is the UX is probably bad (since we cannot guarantee users will give us a directory that is solely for object spilling). What about this? I will always create a directory appended to a given spilling directory and delete that directory directly. For example,

```
file_system_object_spilling_config = {
    "type": "filesystem",
    "params": {
        "directory_path": "/tmp/something"
    }
}
```
This will create a new directory `/tmp/something/ray_spilled_objects/[files]` and I will delete `ray_spilled_objects` so that we can avoid deleting files that are not related to object spilling.